pkg/skaffold/build/tag/git_commit.go

```diff
@@ -94,13 +101,44 @@ func (c *GitCommit) makeGitTag(workingDir string) (string, error) {
 		if c.variant == abbrevCommitSha {
 			args = append(args, "--abbrev-commit")
 		}
+	case treeSha, abbrevTreeSha:
+		gitPath, err := getGitPathToWorkdir(workingDir)
+		if err != nil {
+			return "", err
+		}
+		args = append(args, "rev-parse")
+		if c.variant == abbrevTreeSha {
+			args = append(args, "--short")
+		}
+		// revision must come after the --short flag
+		args = append(args, "HEAD:"+gitPath+"/")
 	default:
 		return "", errors.New("invalid git tag variant: defaulting to 'dirty'")
 	}
 
 	return runGit(workingDir, args...)
 }
 
+func getGitPathToWorkdir(workingDir string) (string, error) {
+	absWorkingDir, err := filepath.Abs(workingDir)
+	if err != nil {
+		return "", err
+	}
+
+	// git reports the gitdir with resolved symlinks, so we need to do this too in order for filepath.Rel to work
+	absWorkingDir, err = filepath.EvalSymlinks(absWorkingDir)
+	if err != nil {
+		return "", err
+	}
+
+	gitRoot, err := runGit(workingDir, "rev-parse", "--show-toplevel")
```

Comment:
`runGit` is working as is in "GenerateFullyQualifiedImageName" function. 
Maybe resolving workspace should be part of ruunGit and not the other way around?

Reply:
hmm looks like the working dir is something else than "git root"
is that correct?

Reply:
@tejal29 `git root` is the top level directory of the git repository, whereas `workingDir` is the top level of the artifact to be tagged. I don't think this functionality can be moved into `runGit`, though. But maybe you have better ideas how to improve the variable naming?

Reply:
maybe a comment for GenerateFullyQualifiedImageName could describe the workingDir is the artifact directory would be more clear?
```
// GenerateFullyQualifiedImageName tags an image with the supplied image name and the git commit.
func (c *GitCommit) GenerateFullyQualifiedImageName(workingDir string, imageName string) (string, error) {
```

Reply:
@tejal29 I added some godoc to the `GenerateFullyQualifiedImageName` interface. Does that help?

Reply:
Yes!